src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForSimulation.cs

```diff
@@ -179,6 +187,13 @@ public IReadOnlyList<Module> FindChangedModules(IMoBiSimulation simulation)
          return simulation.Configuration.ModuleConfigurations.Where(moduleConfiguration => !versionMatch(TemplateModuleFor(moduleConfiguration.Module), moduleConfiguration)).Select(moduleConfiguration => moduleConfiguration.Module).ToList();
       }
 
+      public void CloneSimulation(IMoBiSimulation simulationToClone)
+      {
+         var newSimulation = _cloneManager.CloneSimulation(simulationToClone);
+         InteractionTask.CorrectName(newSimulation, _editTask.GetForbiddenNames(newSimulation, _interactionTaskContext.Context.CurrentProject.Simulations));
```

Comment:
The return value of `InteractionTask.CorrectName()` is ignored, thus if canceled by user - it will crash later (or a simulation with the same name as source will be created).

Also it would be better to get the name for the new simulation first (before cloning) and clone only if not cancelled by user.